apis/python/src/tiledbsoma/indexed_dataframe.py

```diff
@@ -84,7 +84,7 @@ def _create_empty(
             # We need domain=(None,None) for string dims
             lo: Any = None
             hi: Any = None
-            if dtype != str:
+            if dtype != str and dtype != "ascii":
                 if np.issubdtype(dtype, np.integer):
                     lo = np.iinfo(dtype).min
                     hi = np.iinfo(dtype).max - 1
```

Comment:
this came up in the sprint planning meeting today - why rule out other dimension types supported by the underlying database (eg, datetime)?  If `tiledb.Dim` accepts it, seems worth supporting.

Reply:
@bkmartinjr thank you -- not on this PR but on a follow-on -- this `and dtype != "ascii"` should not be on this PR -- I'll remove it. Follow-on tasks include #418 and #419 which are necessarily outside the scope of this PR as the `SOMAReader` will need to be modified to handle non-int.